src/Uno.UI/UI/Xaml/IFrameworkElement.cs

```diff
@@ -208,11 +208,29 @@ public static void InvalidateMeasure(this IFrameworkElement e)
 #endif
 		}
 
-		public static IFrameworkElement FindName(IFrameworkElement e, IEnumerable<View> subviews, string name)
+		private static bool TryFindName(string name, IFrameworkElement element, out object outElement)
 		{
-			if (string.Equals(e.Name, name, StringComparison.Ordinal))
+			if (string.Equals(element.Name, name))
 			{
-				return e;
+				outElement = element;
+				return true;
+			}
+
+			if (element is FrameworkElement fe && fe.Resources[name] is { } resource)
```

Comment:
This is probably not right. Only Resources marked with `x:Name` should be considered by `FindName()`. Consider the following:

```xml
			<Grid>
				<Grid.Resources>
					<SolidColorBrush x:Key="Bronze"
									 Color="RosyBrown" />
					<SolidColorBrush x:Key="Silver"
									 Color="Silver" />
					<SolidColorBrush x:Name="Gold"
									 Color="Gold" />
				</Grid.Resources>
				<TextBlock x:Name="Silver"
						   Text="Silver" />
				<!--<TextBlock x:Name="Gold"
						   Text="Gold" />-->
			</Grid>
```

```csharp
		private void MainPage_Loaded(object sender, RoutedEventArgs e)
		{
			var bronze = this.FindName("Bronze");
			var silver = this.FindName("Silver");
			var gold = this.FindName("Gold");
		}
```

'bronze' should be null, 'silver' should be the TextBlock, and 'Gold' should be the SolidColorBrush.

Reply:
Good catch @davidjohnoliver

Reply:
Something weird is going on when I added a test:

![image](https://user-images.githubusercontent.com/31348972/133500336-39bf9911-da33-4287-a1ae-f8143cdb91d6.png)

In a blank unoapp, the generated code is correct and the app compiles.

Reply:
@davidjohnoliver I thought about using `if (NameScope.GetNameScope(element)?.FindName(name) is { } resource)`, and added a test for the test case you suggested, but the following:

```csharp
			var bronze = this.FindName("Bronze")?.ToString() ?? "<NULL>"; // Should be null
			var silver = this.FindName("Silver")?.ToString() ?? "<NULL>"; // Should be TextBlock
			var gold = this.FindName("Gold")?.ToString() ?? "<NULL>"; // Should be SolidColorBrush

			var bronze2 = myStackPanel.FindName("Bronze")?.ToString() ?? "<NULL>"; // Should be null
			var silver2 = myStackPanel.FindName("Silver")?.ToString() ?? "<NULL>"; // Should be TextBlock
			var gold2 = myStackPanel.FindName("Gold")?.ToString() ?? "<NULL>"; // Should be SolidColorBrush
			ResultsTextBlock.Text = $"{bronze}, {silver}, {gold} - {bronze2}, {silver2}, {gold2}";
```

produced:

![image](https://user-images.githubusercontent.com/31348972/133508138-b86ee48f-a4c3-4c0e-b1c4-730ceb88efa0.png)


(Both `gold` and `gold2` were supposed to be the same, but that didn't happen)

This is expected per the generated code since we call `SetNameScope` only for the page. But not how `FindName` is supposed to work. I'm not sure if NameScopes should somehow be inherited. Especially that on UWP, `Silver.FindName("Gold");` DOES return a SolidColorBrush. The XAML again for reference:

```xml
    <Grid x:Name="grid">
        <Grid.Resources>
            <SolidColorBrush x:Key="Bronze"
									 Color="RosyBrown" />
            <SolidColorBrush x:Key="Silver"
									 Color="Silver" />
            <SolidColorBrush x:Name="Gold"
									 Color="Gold" />
        </Grid.Resources>
        <TextBlock x:Name="Silver" Text="Silver" /> <!-- NOTE: Silver.FindName("Gold") works -->
    </Grid>
```

Reply:
Hmm... Interesting.. There is already `FrameworkPropertyMetadataOptions.Inherits` for `NameScopeProperty`. I don't understand why it's not working as I was expecting.

Reply:
@davidjohnoliver I did try that, but turns out the parent returned is null. No idea why.

Reply:
I'm going to revert those changes for now and open a separate issue to fix the `FindName` issue.

Reply:
Did you do the change? Or are you waiting for the other issue to be fixed ?

Reply:
@jeromelaban I updated the sample to not use `FindName` with an Uno doc comment and filed #7258 to track fixing `FindName`